app/src/main/java/com/gholem/moneylab/features/add/AddFragment.kt

```diff
@@ -1,21 +1,50 @@
 package com.gholem.moneylab.features.add
 
-import android.os.Bundle
-import android.view.LayoutInflater
-import android.view.View
-import android.view.ViewGroup
-import androidx.fragment.app.Fragment
-import com.gholem.moneylab.R
-
-class AddFragment : Fragment() {
-
-    override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_add, container, false)
+import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.gholem.moneylab.arch.base.BaseFragment
+import com.gholem.moneylab.databinding.FragmentAddBinding
+import com.gholem.moneylab.features.add.adapter.AddTransactionsAdapter
+import com.gholem.moneylab.features.add.navigation.AddNavigation
+import com.gholem.moneylab.features.add.viewmodel.AddViewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class AddFragment : BaseFragment<FragmentAddBinding, AddViewModel>() {
+
+    private val viewModel: AddViewModel by viewModels()
+
+    lateinit var addNavigation: AddNavigation
+    lateinit var adapter: AddTransactionsAdapter
+
+    override fun constructViewBinding(): FragmentAddBinding =
+        FragmentAddBinding.inflate(layoutInflater)
+
+    private val dataAdapter: AddTransactionsAdapter by lazy {
+        AddTransactionsAdapter()
     }
 
+
+    override fun init(viewBinding: FragmentAddBinding) {
+        viewModel.init()
+
+        adapter = AddTransactionsAdapter()
+
+        viewBinding.transactionsRecyclerView.adapter = adapter
+        viewBinding.transactionsRecyclerView.layoutManager = LinearLayoutManager(context)
+
+        dataAdapter.setData(viewModel.getMockData())
```

Comment:
And still providing the data before attaching the adapter - please move this line to the end of the method